Kernel/Devices/HID/KeyboardDevice.cpp

```diff
@@ -268,6 +268,9 @@ void KeyboardDevice::key_state_changed(u8 scan_code, bool pressed)
 UNMAP_AFTER_INIT KeyboardDevice::KeyboardDevice()
     : HIDDevice(85, HIDManagement::the().generate_minor_device_number_for_keyboard())
 {
+    auto name = String::formatted("keyboard{}", minor());
+    m_determined_name = KString::try_create(name.substring_view(0));
+    VERIFY(m_determined_name);
```

Comment:
nitpick: I would argue we should just create a factory function create the `KString` there and pass it in to the constructor / initialize later, so it can propagate error instead of `VERIFY`ing allocations. The pattern of allocating in constructors should go away... even if this code is only run on boot, people copy and paste patterns from code and propagate things we shouldn't really be doing.

Reply:
hmm, I agree we shouldn't allocate within the constructor. About the `VERIFY` statement - if we fail to allocate memory for `KString` in this moment, it means we have a bug and the kernel cannot operate normally, so the right thing to do is to assert.

Reply:
> About the `VERIFY` statement - if we fail to allocate memory for `KString` in this moment, it means we have a bug and the kernel cannot operate normally, so the right thing to do is to assert.

I worded my original sentence poorly, I agree we should assert on failure in these cases, we obviously can't continue. I was trying to advocate for moving the assertion + allocation out of the constructor, that's all.